src/utils/sourcemaps.rs

```diff
@@ -947,20 +947,64 @@ impl SourceMapProcessor {
                             bail!("Invalid embedded sourcemap in source file {source_url}");
                         };
 
-                        let mut sourcemap = SourceMap::from_slice(&decoded).with_context(|| {
-                            format!("Invalid embedded sourcemap in source file {source_url}")
-                        })?;
+                        let mut sourcemap =
+                            sourcemap::decode_slice(&decoded).with_context(|| {
+                                format!("Invalid embedded sourcemap in source file {source_url}")
+                            })?;
 
                         let debug_id = sourcemap
-                            .get_debug_id()
+                            .debug_id()
                             .unwrap_or_else(|| inject::debug_id_from_bytes_hashed(&decoded));
 
                         let source_file = self.sources.get_mut(source_url).unwrap();
                         let source_file_contents = Arc::make_mut(&mut source_file.contents);
-                        let adjustment_map = inject::fixup_js_file(source_file_contents, debug_id)
-                            .context(format!("Failed to process {}", source_file.path.display()))?;
 
-                        sourcemap.adjust_mappings(&adjustment_map);
+                        match sourcemap {
+                            DecodedMap::Regular(_) | DecodedMap::Hermes(_) => {
+                                let sourcemap = match &mut sourcemap {
+                                    DecodedMap::Regular(sm) => sm,
+                                    DecodedMap::Hermes(sm) => &mut *sm,
+                                    DecodedMap::Index(_) => {
+                                        unreachable!("SourceMapIndex is handled in other branch")
+                                    }
+                                };
```

Comment:
Matching on a value and then immediately matching on it again inside is very unfortunate. I think I'd rather you just duplicate the code in the `Regular` and `Hermes` cases.

Reply:
That's fair. I suppose I could extract the logic from the match body into a function; that way, we avoid repeating it (otherwise, we'd be repeating the block four times total in this function)